Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/service/impl/ConceptMergeService.java

```diff
@@ -0,0 +1,254 @@
+            ConceptsMergeBean conceptsMergeBean) {
+        for (ConceptEntry entry : conceptEntries) {
+            conceptsMergeBean.setSelectedPosValue(entry.getPos().toLowerCase());
+
+            Set words = createSet(entry.getWord(), conceptsMergeBean.getWords());
+            conceptsMergeBean.setWords(words);
+            conceptsMergeBean.setSelectedListName(entry.getConceptList());
+
+            Set descriptions = createSet(entry.getDescription(), conceptsMergeBean.getDescriptions());
+            conceptsMergeBean.setDescriptions(descriptions);
+
+            Set<String> synonymIds = combineSynonymIds(conceptsMergeBean.getSynonymsids(), entry.getSynonymIds());
+            conceptsMergeBean.setSynonymsids(synonymIds);
+
+            Set<String> equalsSet = createSet(entry.getEqualTo(), conceptsMergeBean.getEqualsValues());
+            conceptsMergeBean.setEqualsValues(equalsSet);
+
+            Set<String> similarSet = createSet(entry.getSimilarTo(), conceptsMergeBean.getSimilarValues());
+            conceptsMergeBean.setSimilarValues(similarSet);
+        }
+
+        // By default new id will be created
+        conceptsMergeBean.setSelectedConceptId("");
+
+        // Removing all the wordnet ids from this list. Users can merge into any
+        // of the user defined CCP id or can create a new id for the merging
+        // concepts. As per design cannot merge into wordnet id, because wordnet
+        // id concepts cannot be updated.
+        Set<String> wordnetSet = new HashSet<>();
+        for (String conceptId : conceptsMergeBean.getConceptIds()) {
+            if (ConceptTypes.SPECIFIC_WORDNET_CONCEPT == conceptTypesService.getConceptTypeByConceptId(conceptId)) {
+                wordnetSet.add(conceptId);
+            }
+        }
+
+        conceptsMergeBean.getConceptIds().removeAll(wordnetSet);
+        conceptsMergeBean.setWordnetIds(wordnetSet);
+        return conceptsMergeBean;
+    }
+
+    @Override
+    public void mergeConcepts(ConceptsMergeBean conceptsMergeBean, String userName)
+            throws LuceneException, IndexerRunningException, IllegalAccessException, DictionaryDoesNotExistException,
+            DictionaryModifyException {
+
+        deleteMergedConcepts(conceptsMergeBean.getConceptIds(), userName,
+                conceptsMergeBean.getSelectedConceptId().trim(), conceptsMergeBean.getSelectedListName());
+
+        if (conceptsMergeBean.getSelectedConceptId().trim().equals("")) {
+            // Add
+            ConceptEntry entry = new ConceptEntry();
+            getConceptEntryFromConceptMergeBean(entry, conceptsMergeBean);
+            conceptManager.addConceptListEntry(entry, userName);
+        } else {
+            // Update
+            ConceptEntry entry = conceptManager.getConceptEntry(conceptsMergeBean.getSelectedConceptId());
+            getConceptEntryFromConceptMergeBean(entry, conceptsMergeBean);
+            conceptManager.storeModifiedConcept(entry, userName);
+        }
```

Comment:
I'd first create a new concept and then delete the old ones. Otherwise, if anything goes wrong (e.g. server crashes right after deletion before creation of new concepts) the old concepts are gone but there are is no new merged concept yet.